pkg/services/ngalert/api/api_prometheus.go

```diff
@@ -28,6 +28,14 @@ type PrometheusSrv struct {
 	store   store.RuleStore
 }
 
+const queryIncludeInternalLabels = "includeInternalLabels"
+
+// grafanaLabels are labels that grafana automatically include as part of the labelset.
+var grafanaLabels = map[string]struct{}{
+	ngmodels.RuleUIDLabel:      {},
+	ngmodels.NamespaceUIDLabel: {},
+}
```

Comment:
I think this list should be in the models and we should protect all APIs: remove from responses, fail requests that have those labels

Reply:
Does https://github.com/grafana/grafana/pull/46548/commits/78d25fb783f4d50dcd0472b21ec08101b70721de address your main concern?

There's still the question of why is `Labels` public on both the `state` and `ngmodels` packages. I took a stab at refactoring it but it was certainly getting out of hand for this PR.

If the main concern about the boundary not being crossed on the API is addressed, I'm happy to follow up on the privatisation of `Labels` for both of these packages in the next PR.

Reply:
Further discussed this with @yuri-tceretian offline -- I misunderstood his original comment but we both realised this is not an issue at this current state.